docs/.vuepress/components/T.vue

```diff
@@ -0,0 +1,342 @@
+<template>
+  <component
+    :is="getTagName"
+    :to="to"
+    :class="`${getLevel} ${getModifiers} ${getColor}`"
+    ><slot
+  /></component>
+</template>
+
+<script>
+export default {
+  name: 'T',
+  props: {
+    // Tag Names
+    h1: { type: Boolean },
+    h2: { type: Boolean },
+    h3: { type: Boolean },
+    h4: { type: Boolean },
+    h5: { type: Boolean },
+    h6: { type: Boolean },
+    p: { type: Boolean },
+    span: { type: Boolean },
+    ul: { type: Boolean },
+    ol: { type: Boolean },
+    li: { type: Boolean },
+    a: { type: Boolean },
+    routerLink: { type: Boolean },
+    div: { type: Boolean },
+
+    // if router link
+    to: { type: String },
+
+    // Type levels
+    l1: { type: Boolean },
+    l2: { type: Boolean },
+    l3: { type: Boolean },
+    l4: { type: Boolean },
+    l5: { type: Boolean },
+    l6: { type: Boolean },
+    l7: { type: Boolean },
+    l8: { type: Boolean },
+
+    // Colors
+    cPrimary: { type: Boolean },
+    cText: { type: Boolean },
+
+    // Shades
+    s100: { type: Boolean },
+    s200: { type: Boolean },
+    s300: { type: Boolean },
+    s400: { type: Boolean },
+    s500: { type: Boolean },
+    s600: { type: Boolean },
+    s700: { type: Boolean },
+    s800: { type: Boolean },
+    s900: { type: Boolean },
+
+    // modifiers
+    //// text center
+    center: { type: Boolean },
+    //// no margins
+    ma0: { type: Boolean },
+    //// no margin-bottom
+    mb0: { type: Boolean },
+    //// no margin-top
+    mt0: { type: Boolean },
+    //// widths
+    w100: { type: Boolean }
+  },
+  computed: {
+    getTagName() {
+      return (
+        (this.h1 && 'h1') ||
+        (this.h2 && 'h2') ||
+        (this.h3 && 'h3') ||
+        (this.h4 && 'h4') ||
+        (this.h5 && 'h5') ||
+        (this.h6 && 'h6') ||
+        (this.p && 'p') ||
+        (this.span && 'span') ||
+        (this.ul && 'ul') ||
+        (this.ol && 'ol') ||
+        (this.li && 'li') ||
+        (this.a && 'a') ||
+        (this.routerLink && 'router-link') ||
+        (this.div && 'div') ||
+        // default to paragraph
+        'p'
+      )
+    },
+    getLevel() {
+      return (
+        (this.l1 && 'l1') ||
+        (this.l2 && 'l2') ||
+        (this.l3 && 'l3') ||
+        (this.l4 && 'l4') ||
+        (this.l5 && 'l5') ||
+        (this.l6 && 'l6') ||
+        (this.l7 && 'l7') ||
+        (this.l8 && 'l8') ||
+        // default to inherit
+        ''
+      )
+    },
+
+    getModifiers() {
```

Comment:
Seems these props could be replaced by just passing classnames, e.g. 
```<T p class="subtitle mw55ch s300 p l4 c-text">```
instead of:
```<T p l4 cText s300 class="subtitle mw55ch">```
since `getModifiers()` just adds those classes.